### ISSUE TITLE
Add vertical plots

### ISSUE THREAD

**Issue by EdwardSafford-NOAA** [NOAA-EMC/PyGSI#58]
Extend pyGSI/plot_diags.py by adding a vertical plot routine.  This is intended to support gps data (class conventional) but may be useful to other types as well.

**Comment by kevindougherty-noaa** [comment]
This would be a great feature. I plan to come out with a PyGSI version 2 with more updated features/better written and that will utilize the new package we have been working on `emcpy` . A vertical plot routine would be great to add to `emcpy.plots`.

https://github.com/NOAA-EMC/emcpy
https://noaa-emc.github.io/emcpy/emcpy.html

**Comment by EdwardSafford-NOAA** [comment]
Getting there:

![2021060518_conv_gps_o-f_265_vertical(3)](https://user-images.githubusercontent.com/62339196/129944633-f6731752-b90d-417a-8a1b-435e6aac4053.png)

**Comment by kevindougherty-noaa** [comment]
Coming together nicely! One suggestion just visually, I would keep the RMSE and obscount text black and add a legend that says the blue line is obscount and red line is RMSE.

**Comment by EdwardSafford-NOAA** [comment]
Ok, like this then:

![2021060518_conv_gps_o-f_265_vertical(4)](https://user-images.githubusercontent.com/62339196/129954619-18063e33-0505-486f-bac5-797ffbeca9c7.png)

**Comment by kevindougherty-noaa** [comment]
I was thinking something more along the lines like this: 
![image](https://user-images.githubusercontent.com/69815622/129956107-c07ce73b-e8fc-4958-a291-24df10003df1.png)

All you need to do is add labels when plotting the lines and then add a legend. (https://www.pythonprogramming.in/plot-line-graph-with-multiple-lines-with-label-and-legend.html)

**Comment by EdwardSafford-NOAA** [comment]
Cool!

**Comment by EdwardSafford-NOAA** [comment]
@kevindougherty-noaa @CoryMartin-NOAA 

I've successfully tested vertical plots now for omf, oma, observation, and hofx with analysis_use = False, and I've moved on to handling the anaysis_use = True case.  I was able to produce plots for assimilated and monitored data but I think the way the data dict is organized by the pyGSI/diag.py/get_data() routine is less than optimal when analysis_use = true.

When analysis_use is set to True the data dict returned by get_data is ordered [levels][assim|mon] or [levels][u|v][assim|mon].  Is there a reason the level is first in the dict structure?  In order to keep my vertical plot routine generic I found I needed to re-arrange the data dict into [assim|mon][levels] so I could step through the dict by the first key sending all assimilated data in one call and then all monitored data in the next.  The other option was to have the plot routine step through the unmodified data dict and pull out either assimilated or monitored data, but that seemed awkward.  Is there a reason for the [levels][assim|mon] order that I'm missing?  If not I'd like to rewrite the pyGSI/diag.py/get_data() function to reverse the order of levels and assim|mon.  If there is a good reason to keep it as is then I'll add a function to diag.py to reorder the data dict which could be called after get_data().  Thoughts?

**Comment by kevindougherty-noaa** [comment]
@EdwardSafford-NOAA The plot looks great!

At the moment, I am not sure why levels were chosen before assimilated and monitored data. Within the next few weeks, I plan to do a pretty significant redevelopment of `PyGSI` to clean things up, use new approaches I've learned, and implement `emcpy` routines to the scripts (especially plotting).  What most likely will happen is that we can implement your vertical plotting technique into emcpy so that it is generic for other groups to reproduce with different datasets and then have a specific script (likely `plot_diags.py`) that handles GSI data.

**Comment by EdwardSafford-NOAA** [comment]
@kevindougherty-noaa  

Ok I'll leave the data order as is for now and verify that vertical plots work for all conventional data types with analysis_use in both states.  Once your redevelopment is complete we can figure out how/where my stuff fits.

**Comment by EdwardSafford-NOAA** [comment]
@kevindougherty-noaa I forked and cloned emcpy but ran into a problem.  Per the README I ran >pip install .
This was the response:

ERROR: Could not install packages due to an EnvironmentError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Max retries exceeded with url: /packages/16/77/44ad4ff98f944a743735575dd6345959e19b43984d495b4738e309452a20/netCDF4-1.5.7-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (Caused by NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7f6f8b516bd0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Any experience with that issue heretofore?

**Comment by CoryMartin-NOAA** [comment]
@EdwardSafford-NOAA which machine? If Hera, this is expected because it is annoying with external connections.

**Comment by EdwardSafford-NOAA** [comment]
@CoryMartin-NOAA yes Hera.  If there's no way around it I guess I'll move to wcoss.

**Comment by CoryMartin-NOAA** [comment]
@aerorahul do you have any insight on how best to handle this? Should @EdwardSafford-NOAA wait until we have a pyvenv on Hera? Or set up a conda env and use that?

**Comment by EdwardSafford-NOAA** [comment]
This issue has been largely resolved by updates to the emcpy package.  It's not clear at this point that there is a need for 'vertical' plots that isn't currently fulfilled by existing line plot capabilities.  If that changes I'll be glad to open a new issue.
